### PR TITLE
Add support for hooking CanBeAutobalanced to SDKHooks.

### PR DESCRIPTION
**Pull request by psychonic**
Not many games support this, but it is in the SDK (and thus likely in many mods). This is also already an often used feature in TF2, provided by an unsupported extension that breaks often (using a different method to hook).

I don't know offhand of other games/mods that support this, but it's trivial to add the offset when finding one.

### CODE REVIEW

File: extensions/sdkhooks/extension.cpp [hunk 1]
```diff
@@ -905,6 +913,46 @@ bool SDKHooks::Hook_LevelInit(char const *pMapName, char const *pMapEntities, ch
 /**
  * CBaseEntity Hook Handlers
  */
+bool SDKHooks::Hook_CanBeAutobalanced()
+{
+	CBaseEntity *pPlayer = META_IFACEPTR(CBaseEntity);
+	int entity = gamehelpers->EntityToBCompatRef(pPlayer);
+
+	bool origRet = SH_MCALL(pPlayer, CanBeAutobalanced)();
```
Comment (KyleSanderson) [inline extensions/sdkhooks/extension.cpp]: should be inside the hook before the pawn calls.

File: extensions/sdkhooks/extension.cpp [hunk 2]
```diff
@@ -905,6 +913,46 @@ bool SDKHooks::Hook_LevelInit(char const *pMapName, char const *pMapEntities, ch
...
+			// (so if multiple plugins returning different answers,
+			//  the one(s) that changed it win)
+			if ((res != 0) != origRet)
+				newRet = !origRet;
+		}
+
+		break;
+	}
+
+	RETURN_META_VALUE(MRES_SUPERCEDE, newRet);
```
Comment (KyleSanderson) [inline extensions/sdkhooks/extension.cpp]: use ignore here.

File: extensions/sdkhooks/extension.cpp [hunk 3]
```diff
@@ -905,6 +913,46 @@ bool SDKHooks::Hook_LevelInit(char const *pMapName, char const *pMapEntities, ch
...
+			callback->Execute(&res);
+
+			// Only update our new ret if different from original
+			// (so if multiple plugins returning different answers,
+			//  the one(s) that changed it win)
+			if ((res != 0) != origRet)
+				newRet = !origRet;
+		}
+
+		break;
```
Comment (KyleSanderson) [inline extensions/sdkhooks/extension.cpp]: if override takes precedence, RETURN_META_VALUE MRES_SUPERCEDE here instead.

**Comment by KyleSanderson** [comment]
No one else is probably hooking these, nits/hook citizen issues aside it seems absolutely fine.

File: extensions/sdkhooks/extension.cpp [hunk 4]
```diff
@@ -905,6 +913,46 @@ bool SDKHooks::Hook_LevelInit(char const *pMapName, char const *pMapEntities, ch
...
+		if (vhook != vtablehooklist[entry]->vtablehook)
+		{
+			continue;
+		}
+
+		ke::Vector<IPluginFunction *> callbackList;
+		PopulateCallbackList(vtablehooklist[entry]->hooks, callbackList, entity);
+		for (entry = 0; entry < callbackList.length(); ++entry)
+		{
+			cell_t res;
```
Comment (KyleSanderson) [inline extensions/sdkhooks/extension.cpp]: initialize res incase there's a plugin error with forwardsys (pretty sure every other hook does this).

**Comment by psychonic** [comment]
@KyleSanderson Thanks for fixing this up.

**Pull request merged**